File: src/ByteBite.Infrastructure/Persistence/Entities/Conversation.cs
using System;
using System.Collections.Generic;

namespace ByteBite.Infrastructure.Persistence.Entities;

public partial class Conversation
{
    public Guid Id { get; set; }

    public Guid OrderId { get; set; }

    public Guid StoreId { get; set; }

    public Guid? CustomerId { get; set; }

    public string? DeviceId { get; set; }

    public DateTime LastMessageAt { get; set; }

    public int CustomerUnreadCount { get; set; }

    public int MerchantUnreadCount { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public virtual Customer? Customer { get; set; }

    public virtual Order Order { get; set; } = null!;

    public virtual Store Store { get; set; } = null!;

    public virtual ICollection<ConversationMessage> Messages { get; set; } = new List<ConversationMessage>();
}

File: web/src/types/models/conversation.ts
export interface ConversationOrderSummaryDto {
  id: string
  orderNo: string
  pickupCode: string
  status: string
  diningMode: string
  tableNo?: string
  totalAmount: number
  actualAmount: number
  createdAt: string
}

export interface ConversationCustomerSummaryDto {
  id: string
  phone?: string
  username?: string
  nickname?: string
}

export interface ConversationDto {
  id: string
  orderId: string
  storeId: string
  storeName?: string
  storeCode?: string
  customerId?: string
  deviceId?: string
  lastMessageAt: string
  customerUnreadCount: number
  merchantUnreadCount: number
  order?: ConversationOrderSummaryDto
  customer?: ConversationCustomerSummaryDto
}

export interface ConversationMessageDto {
  id: string
  conversationId: string
  senderType: 'customer' | 'merchant'
  senderId?: string
  content: string
  messageType: string
  readAt?: string
  createdAt: string
}

export interface StartConversationRequest {
  customerId?: string
  deviceId?: string
}

export interface SendConversationMessageRequest {
  senderType: 'customer' | 'merchant'
  senderId?: string
  content: string
  storeId?: string
  orderId?: string
}

export interface ConversationEventPayload {
  conversationId: string
  message: ConversationMessageDto
  conversation?: ConversationDto
  unreadCount?: number
}

export interface ConversationUnreadChangedPayload {
  scope: 'customer' | 'merchant'
  storeId?: string
  customerId?: string
  deviceId?: string
  count: number
}
